src/opm/output/eclipse/AggregateActionxData.cpp

```diff
@@ -574,23 +574,25 @@ const std::map<cmp_enum, int> cmpToIndex = {
 
                 //Treat well, group and field left hand side conditions
                 if (it_lhsq != lhsQuantityToIndex.end()) {
-                    std::string wn = "";
                     //Well variable
                     if (it_lhsq->first == "W") {
                         //find the well that violates action if relevant
-                        for (const auto& well : wells)
-                        {
-                            if (ar.has_well(well.name())) {
-                                //set well name
-                                wn = well.name();
-                                break;
+                        if (ar) {
+                            std::optional<std::string> wn;
+                            for (const auto& well : wells)
+                            {
+                                if (ar.has_well(well.name())) {
+                                    //set well name
+                                    wn = well.name();
+                                    break;
+                                }
                             }
```

Comment:
Since you're changing this entire section anyway, I'd suggest pulling it out into a helper function to reduce the indentation level and increase readability.  Furthermore, the loop is essentially a `std::find_if()` from `<algorithm>`.  You don't really need an `optional<string>` to capture that result when
```C++
pos != wells.end()
```
will answer the same question.

Reply:
I have not extracted it out into a separate function, but it is refactored to use `std::find_if`as suggested.